### PR TITLE
Create header component

### LINKED ISSUE #226: Create header component
### Description
Create header component for home page

### Requirements

- component must be responsible
- dark mode ready
- should include transition visuals
- include active, focus shadows and effects

### Deliverables

- should have color in color 
- main logo component
- listings menu with 3 labels and search icon
- profile menu will be implemented in task #237

### PR DESCRIPTION
**Pull request by Ozodbek4**
Closes #226

### CODE REVIEW

File: src/Training.AirBnb.Clone.Frontend/src/main.ts [hunk 1]
```diff
@@ -1,14 +1,6 @@
...
 import App from './App.vue'
-import router from './infrastructure/router'
-
-const app = createApp(App)
-
-app.use(createPinia())
-app.use(router)
 
-app.mount('#app')
+createApp(App).mount('#app')
```
Comment (botirjonabdumannonovv) [inline src/Training.AirBnb.Clone.Frontend/src/main.ts:6]: tepadagilarni nimaga remove qilib yubordiz.
ustoz route qo'shganda shuni ham qo'shing degan edi

**Review by botirjonabdumannonovv — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — DISMISSED** [review]

**Review by Xolmuratov02 — DISMISSED** [review]

**Review by Abdurahmon0412 — APPROVED** [review]

File: src/Training.AirBnb.Clone.Frontend/src/infrastructure/services/AppThemeService.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        return window.matchMedia("(prefers-color-scheme: dark)").matches
+    }
+
+    /**
+     * Toggles the dark mode of the application.
+     * It adds or removes the "dark" class from the document body and updates the darkMode value in localStorage.
+     */
+    public toggleDarkMode(): void {
+        document.body.classList.toggle("dark");
+        const dakMode= localStorage.getItem("darkMode") !== null ? localStorage.getItem("darkMode") == "true" : false;
```
Comment (Xolmuratov02) [inline src/Training.AirBnb.Clone.Frontend/src/infrastructure/services/AppThemeService.ts:23]: O'zgaruvchi nomi dak mode bo'lib qolibdi

**Review by Xolmuratov02 — CHANGES_REQUESTED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Xolmuratov02 — DISMISSED** [review]

File: src/Training.AirBnb.Clone.Frontend/index.html [hunk 3]
```diff
@@ -3,10 +3,12 @@
   <head>
     <meta charset="UTF-8">
     <link rel="icon" href="/favicon.ico">
+    <script type="module" src="/src/main.ts"></script>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <script type="module" src="/src/main.ts"></script>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/index.html:8]: Why 2 sources files are added ? Also need to add script as last element of body

File: src/Training.AirBnb.Clone.Frontend/src/App.vue [hunk 4]
```diff
@@ -1,3 +1,11 @@
 <template>
 
-</template>
\ No newline at end of file
+  <!--    <main-logo/>-->
+
+  <home-view/>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/App.vue:5]: Remove unnecessary empty space

File: src/Training.AirBnb.Clone.Frontend/src/common/views/HomeView.vue [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  <button class="mt-80" @click="appThemeService.toggleDarkMode()">Dark mode</button>
+</template>
+
+<script setup lang="ts">
+
+import MainHeader from "@/common/components/MainHeader.vue";
+import { AppThemeService } from "@/infrastructure/services/AppThemeService";
+
+// import LocationsContainer from "@/modules/locations/components/ListingsContainer.vue";
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/common/views/HomeView.vue:15]: Remove comments

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request merged**